### PR TITLE
 Support automatic reload on file change #43

### PR DESCRIPTION
**Pull request by cag**
https://github.com/timothycrosley/hug/issues/43

### CODE REVIEW

**Comment by timothycrosley** [comment]
Very very cool! I know this will make a lot of people happy when merged in and released, I'll take a look at it tonight

**Comment by cag** [comment]
Thanks! Sorry I've been silent: I've just been busy. I've been meaning to do a bit of refactoring and I also wanted to make it more optional. I know somebody who is unable to install watchdog on his machine, so I didn't want to make that a hard dependency for this project.

File: hug/api.py [hunk 1]
```diff
@@ -209,18 +182,6 @@ def documentation(self, base_url=None, api_version=None):
         documentation['handlers'] = version_dict
         return documentation
 
-    def serve(self, port=8000, no_documentation=False):
```
Comment (timothycrosley) [inline hug/api.py]: unfortunately, this pull request substantially changes the API behind how hug servers are started manually - a function used by many people. This makes it hard for me to accept the pull request as-is

**Comment by cag** [comment]
Aw... I'm actually currently looking at the Python module reloading in more detail. If the serve method needs to stay, I'm going to have to refactor this a bit more. IIRC I didn't remove the server() method, but if the serve() method is important, there are still ways to make this happen. Unfortunately, that would mean either mean muddying up the api.py file a bit with watchdog dependencies or doing some sort of convoluted dependency injection. If you have any recommendations, I'll gladly look into them.

**Comment by cag** [comment]
This has mysteriously stopped working on my computer, and is not working on some machines. I'm trying to patch reload() to work for this, but right now, maybe the best way is to kill and restart the whole process with something external.

**Pull request closed without merging**